packages/core/src/repository/data-source/array-network.data-source.provider.ts

```diff
@@ -0,0 +1,49 @@
+import { ApiRequestService } from "./api-request.service";
+import { Type } from "../../helpers";
+import { DataSource, GetDataSource, PutDataSource } from "./data-source";
+import { NetworkDataSource } from "./network.data-source";
+import { DataSourceMapper, GetDataSourceMapper, PutDataSourceMapper } from "./data-source-mapper";
+import { ArrayMapper, BlankMapper, JsonDeserializerMapper } from "../mapper/mapper";
+
+export function provideGetArrayNetworkDataSource<T extends unknown | void>(
```

Comment:
What do you mean by `T extends unknown | void`? Isn't this the same as just `T`? Also, regarding the `void`, does `GetDataSource<void[]>` make sense? We can probably leave it as:

```suggestion
export function provideGetArrayNetworkDataSource<T>(
```

Reply:
@doup The thing is that the generic type should allow void as a type, and having T you cannot create a GetDataSource<void>. I just wanted to allow this because we allow calls to the backend that does not need any response. Since this is a generic library, I didn't want to restrict the usage because the API might not be ours and might return nothing at some endpoint. What do you think?

Reply:
Makes sense to me